src/App.tsx

```diff
@@ -8,54 +8,232 @@ import { PostsList } from './components/PostsList';
+  } = apiErrors;
+
+  const deleteCommentFromList = useCallback((deletedId: number) => {
+    setCcomments(prevComments => prevComments.filter(comment => comment.id
+      !== deletedId));
+  }, []);
+
+  const addCommentToList = (newComment: Comment) => {
+    setCcomments(prevComments => [...prevComments, newComment]);
+  };
+
+  const removeComment = (commentId: number) => {
+    deleteCommentFromList(commentId);
+    deleteComment(commentId)
+      .catch(() => {
+      });
+  };
+
+  const handleSelectUser = (user: User) => {
+    if (selectedUser?.id !== user.id) {
+      setSelectedPost(null);
+    }
+
+    setSelectedUser(user);
+  };
+
+  const handleSelectPost = (post: Post | null) => {
+    if (selectedPost?.id === post?.id) {
+      setSelectedPost(null);
+
+      return;
+    }
+
+    setSelectedPost(post);
+    setIsFormVisible(false);
+  };
+
+  const loadUsers = async () => {
+    try {
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch {
+      setApiErrors(prevState => ({ ...prevState, userError: true }));
+    }
+  };
+
+  const loadPosts = async () => {
+    if (selectedUser) {
+      setIsUserLoading(true);
+
+      setApiErrors(prevState => ({ ...prevState, postError: false }));
+      try {
+        const postsFromServer = await getPosts(selectedUser.id);
+
+        setPosts(postsFromServer);
+      } catch {
+        setApiErrors(prevState => ({ ...prevState, postError: true }));
+      } finally {
+        setIsUserLoading(false);
+      }
+    }
+  };
+
+  const loadComments = async () => {
+    if (selectedPost) {
+      setApiErrors(prevState => ({ ...prevState, commentsError: false }));
+
+      setIsCommentsLoading(true);
+      setCcomments([]);
+
+      try {
+        const commentsFromServer = await getComments(selectedPost.id);
+
+        setCcomments(commentsFromServer);
+      } catch {
+        setApiErrors(prevState => ({ ...prevState, commentsError: true }));
+      } finally {
+        setIsCommentsLoading(false);
+      }
+    }
+  };
+
+  useEffect(() => {
+    loadUsers();
+  }, []);
+
+  useEffect(() => {
+    loadPosts();
+  }, [selectedUser]);
+
+  useEffect(() => {
+    loadComments();
+  }, [selectedPost]);
+
+  const isUserError = userError;
+  const isNoSelectedUser = !selectedUser && !userError;
+  const isLoadingPosts = isUserLoading;
+  const isNoPostsYet = !isUserLoading && !posts?.length;
+  const hasPosts = !isUserLoading && !!posts?.length;
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  onUserSelect={handleSelectUser}
+                />
               </div>
```

Comment:
Not fixed
![2023-09-29_15h15_46](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/61904995/c45dff56-b6c2-4df3-b207-5dc5f14cc156)